src/io/ResultWriter.cpp

```diff
@@ -1,88 +1,86 @@
-			<< std::setw(_writePrecision+15) << "beta_trans"
-			<< std::setw(_writePrecision+15) << "beta_rot"
-			<< std::setw(_writePrecision+15) << "c_v"
-			<< std::setw(_writePrecision+15) << "N"
+		const std::string resultfile(_outputPrefix+".res");
+		std::ofstream resultStream;
+		resultStream.open(resultfile.c_str(), std::ios::out);
+		resultStream << std::setw(10) << "timestep"
+			<< std::setw(_writeWidth) << "time"
+			<< std::setw(_writeWidth) << "U_pot_avg"
+			<< std::setw(_writeWidth) << "U_kin_avg"
+			<< std::setw(_writeWidth) << "U_kinTrans_avg"
+			<< std::setw(_writeWidth) << "U_kinRot_avg"
+			<< std::setw(_writeWidth) << "p_avg"
+			<< std::setw(_writeWidth) << "c_v"
+			<< std::setw(_writeWidth) << "N"
 			<< std::endl;
+		resultStream.close();
 	}
 }
 
 void ResultWriter::endStep(ParticleContainer *particleContainer, DomainDecompBase *domainDecomp, Domain *domain,
                            unsigned long simstep) {
 
-	// Writing of cavities now handled by CavityWriter
+	const uint64_t globalNumMolecules = domain->getglobalNumMolecules(true, particleContainer, domainDecomp);
 
-	unsigned long globalNumMolecules = domain->getglobalNumMolecules(true, particleContainer, domainDecomp);
-	double cv = domain->cv();
+	_uPot_acc += domain->getGlobalUpot();
+	_p_acc += domain->getGlobalPressure();
+	_uKinTrans_acc += domain->getGlobalUkinTrans();
+	_uKinRot_acc += domain->getGlobalUkinRot();
 
-	_U_pot_acc->addEntry(domain->getGlobalUpot());
-	_p_acc->addEntry(domain->getGlobalPressure());
-	if((domainDecomp->getRank() == 0) && (simstep % _writeFrequency == 0)){
-		_resultStream << std::setw(10) << simstep << std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _simulation.getSimulationTime()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalUpot()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _U_pot_acc->getAverage()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalPressure()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _p_acc->getAverage()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalBetaTrans()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalBetaRot()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << cv
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << globalNumMolecules
-			<< std::endl;
-	}
-}
+	_numSamples++;
 
-void ResultWriter::finish(ParticleContainer * /*particleContainer*/,
-						  DomainDecompBase *domainDecomp, Domain * /*domain*/){
+	if ((simstep % _writeFrequency == 0) and (simstep > 0UL)) {
```

Comment:
Yes, in theory, you are right. But as far as I know, this is almost nowhere done.
In addition, you can also just enter a negative number and you will never notice, as it will be converted to a (really huge) unsigned one. This could be solved by using a signed integer for `_writeFrequency` and checking for `<=0` but it would be better to implement those parsing checks more fundamentally. The whole input parsing/sanity check does not really exist (only for boolean now).
A possible solution could start [here](https://github.com/ls1mardyn/ls1-mardyn/blob/981cddfbf729484d9b2e30adb4019de07b166076/src/utils/xmlfile.cpp#L531-L608); maybe by adding more explicit functions (and checks).

Reply:
you're right, we should have more comprehensive input checks. It's just that normally a frequency 0 in any writing function should mean that nothing should be written, so it's not unreasonable for someone to assume that 0 is valid input. I think the check you've included is fine here.